tests/core/gas-strategies/test_time_based_gas_price_strategy.py

```diff
@@ -139,3 +142,56 @@ def test_time_based_gas_price_strategy(strategy_params, expected):
     w3.eth.setGasPriceStrategy(time_based_gas_price_strategy)
     actual = w3.eth.generateGasPrice()
     assert actual == expected
+
+
+@pytest.mark.parametrize(
+    'strategy_params_zero,expected_exception_message',
+    (
+        # 120 second wait times, 0 sample_size
+        (dict(max_wait_seconds=80, sample_size=0, probability=98),
+            'Constrained sample size is 0'),
+        (dict(max_wait_seconds=80, sample_size=0, probability=90),
+            'Constrained sample size is 0'),
+        (dict(max_wait_seconds=80, sample_size=0, probability=50),
+            'Constrained sample size is 0'),
+        # 60 second wait times, 0 sample_size
+        (dict(max_wait_seconds=60, sample_size=0, probability=98),
+            'Constrained sample size is 0'),
+        (dict(max_wait_seconds=60, sample_size=0, probability=90),
+            'Constrained sample size is 0'),
+        (dict(max_wait_seconds=60, sample_size=0, probability=50),
+            'Constrained sample size is 0'),
+        # 40 second wait times, 0 sample_size
+        (dict(max_wait_seconds=40, sample_size=0, probability=98),
+            'Constrained sample size is 0'),
+        (dict(max_wait_seconds=40, sample_size=0, probability=90),
+            'Constrained sample size is 0'),
+        (dict(max_wait_seconds=40, sample_size=0, probability=50),
+            'Constrained sample size is 0'),
+        # 20 second wait times, 0 sample_size
+        (dict(max_wait_seconds=20, sample_size=0, probability=98),
+            'Constrained sample size is 0'),
+        (dict(max_wait_seconds=20, sample_size=0, probability=90),
+            'Constrained sample size is 0'),
+        (dict(max_wait_seconds=20, sample_size=0, probability=50),
+            'Constrained sample size is 0'),
+    ),
```

Comment:
Hah, maybe a little overkill with all these different other parameter values, but it's not worth holding up the PR to tighten this up.